edge-modules/api-proxy-module/src/monitors/certs_monitor.rs

```diff
@@ -128,6 +118,25 @@ pub fn start(
     Ok((monitor_loop, shutdown_handle))
 }
 
+async fn check_server_cert(cert_monitor: &mut CertificateMonitor) -> Result<bool, Error> {
```

Comment:
Since this is getting pulled out into a separate method should we add a UT?

Reply:
Possible but would need to inject the method "file::write_binary_to_file"